lib/improver/spotdata/main.py

```diff
@@ -64,82 +62,84 @@ def run_spotdata(config_file_path, data_path, ancillary_path,
-        Contains:
-
-        latitudes : list of ints/floats or None
-            A list of latitudes for running on the fly for a custom set of
-            sites. The order should correspond to the subsequent latitudes and
-            altitudes variables to construct each site.
-
-        longitudes : list of ints/floats or None
-            A list of longitudes for running on the fly for a custom set of
-            sites.
-
-        altitudes : list of ints/floats or None
-            A list of altitudes for running on the fly for a custom set of
-            sites.
-
-        site_ids : list of ints or None
-            A list of site_ids to associate with the above on the fly
-            constructed sites. This must be ordered the same as the latitudes/
-            longitudes/altitudes lists.
-
-    forecast_date : string (YYYYMMDD)
-        A string of format YYYYMMDD defining the start date for which forecasts
-        are required.
-
-    forecast_time : integer
-        An integer giving the hour on the forecast_date at which to start the
-        forecast output; 24hr clock such that 17 = 17Z for example.
-
-    forecast_length : integer
-        An integer giving the desired length of the forecast output in hours
-        (e.g. 48 for a two day forecast period). Defaults to 168 (7 days).
-
-    output_path : string
-        Path to which output file containing processed diagnostic should be
-        written.
-
-    use_multiprocessing : boolean
-        A switch determining whether to use multiprocessing in the data
-        extraction step.
+        diagnostics : dict
+            Dictionary containing the information regarding the methods that
+            will be applied for a specific diagnostic, as well as the data
+            following loading in a cube, and any additional data required to
+            be able to compute the methods requested.
+
+            For example:
+            {
+                "temperature": {
+                    "diagnostic_name": "air_temperature",
+                    "extrema": True,
+                    "filepath": "temperature_at_screen_level",
+                    "interpolation_method": "use_nearest",
+                    "neighbour_finding": {
+                        "land_constraint": False,
+                        "method": "fast_nearest_neighbour",
+                        "vertical_bias": None
+                    "data": iris.cube.CubeList
+                    "additional_data" : iris.cube.CubeList
+                    }
+                }
+            }
+
+        ancillary_data : dict
+            Dictionary containing named ancillary data; the key gives the name
+            and the item is the iris.cube.Cube of data.
+
+        sites : dict
+            Contains:
+
+            latitudes : list of ints/floats or None
+                A list of latitudes for running on the fly for a custom set of
+                sites. The order should correspond to the subsequent latitudes
```

Comment:
I think that this isn't just for running on the fly but also for sites read from a file now. We probably need to remove mention about "on the fly"